packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -117,6 +99,61 @@ export const getLetterEditingOptions = <TDate>(
   }
 };
 
+export const cleanDigitSectionValue = <TDate>(
+  utils: MuiPickersAdapter<TDate>,
+  value: number,
+  sectionType: FieldSectionType,
+  format: string,
+  hasLeadingZeros: boolean,
+  sectionBoundaries: FieldSectionValueBoundaries<TDate, any>,
+) => {
+  const hasLetterPrefix = () => {
+    const startOfYear = utils.startOfYear(utils.date()!);
+    const startOfYearStr = utils.formatByString(startOfYear, format);
+
+    return Number.isNaN(Number(startOfYearStr));
+  };
+
+  if (process.env.NODE_ENV !== 'production') {
+    if (sectionType !== 'day' && hasLetterPrefix()) {
+      throw new Error(
+        [
+          `MUI: The token "${format}" is a digit format with a letter prefix.'
+             This type of format is only supported for 'day' sections`,
+        ].join('\n'),
+      );
+    }
+  }
+
+  if (sectionType === 'day' && hasLetterPrefix()) {
+    const date = utils.setDate(
+      (sectionBoundaries as FieldSectionValueBoundaries<TDate, 'day'>).longestMonth,
+      value,
+    );
+    return utils.formatByString(date, format);
+  }
+
+  // queryValue without leading `0` (`01` => `1`)
+  const valueStr = value.toString();
+
+  if (hasLeadingZeros) {
+    const size = utils.formatByString(utils.date()!, format).length;
+    let cleanValueStr = valueStr;
+
+    // We remove the leading zeros
```

Comment:
The `We` part does not seem to add anything relevant. 🤷 
```suggestion
    // Remove the leading zeros
```